contracts/pool/BundleService.sol

```diff
@@ -233,6 +233,103 @@ contract BundleService is
         bundleManager.lock(bundleNftId);
     }
 
+    /// @inheritdoc IBundleService
+    function stake(
+        IInstance instance,
+        NftId bundleNftId, 
+        Amount amount
+    ) 
+        external 
+        virtual
+        // TODO: restricted() (once #462 is done)
+    {
+        IBundle.BundleInfo memory bundleInfo = instance.getInstanceReader().getBundleInfo(bundleNftId);
+        StateId bundleState = instance.getInstanceReader().getMetadata(bundleNftId.toKey32(BUNDLE())).state;
+
+        if(bundleState != ACTIVE() || bundleInfo.expiredAt < TimestampLib.blockTimestamp() || bundleInfo.closedAt.gtz()) {
```

Comment:
staking should be possible for locked bundles too (ie if bundle is not closed or expired)

Reply:
uh oh i missed that locked bundles are in PAUSED state. 😊 fixed and tested.